packages/bee-q/src/components/option-list/bq-option-list.tsx

```diff
@@ -0,0 +1,118 @@
+
+  // Prop lifecycle events
+  // =======================
+
+  // Events section
+  // Requires JSDocs for public API documentation
+  // ==============================================
+
+  /** Handler to be called when `bq-option` item loses focus. */
+  @Event() bqBlur: EventEmitter<HTMLBqOptionElement>;
+
+  /** Handler to be called when `bq-option` item gets focus. */
+  @Event() bqFocus: EventEmitter<HTMLBqOptionElement>;
+
+  /** Handler to be called when `bq-option` is selected (on click/enter press). */
+  @Event() bqSelect: EventEmitter<HTMLBqOptionElement>;
+
+  // Component lifecycle events
+  // Ordered by their natural call order
+  // =====================================
+
+  // Listeners
+  // ==============
+
+  @Listen('bqOptionBlur', { passive: true })
+  onBqOptionBlur(event: CustomEvent<HTMLElement>) {
+    const { target: item } = event;
+    if (isHTMLElement(item, 'bq-option')) this.bqBlur.emit(item);
+  }
+
+  @Listen('bqOptionFocus', { passive: true })
+  onBqOptionFocus(event: CustomEvent<HTMLElement>) {
+    const { target: item } = event;
+    if (isHTMLElement(item, 'bq-option')) this.bqFocus.emit(item);
+  }
+
+  @Listen('bqOptionClick', { passive: true })
+  @Listen('bqOptionEnter', { passive: true })
+  onBqSelect(event: CustomEvent<HTMLElement>) {
+    const { target: item } = event;
+
+    this.optionItems.forEach((option: HTMLBqOptionElement) => (option.selected = option === item));
+    if (isHTMLElement(item, 'bq-option')) this.bqSelect.emit(item);
+  }
+
+  // Public methods API
+  // These methods are exposed on the host element.
+  // Always use two lines.
+  // Public Methods must be async.
+  // Requires JSDocs for public API documentation.
+  // ===============================================
+
+  // Local methods
+  // Internal business logic.
+  // These methods cannot be called from the host element.
+  // =======================================================
+
+  private setSlotElem = () => {
+    this.slot = this.baseElem.querySelector('slot');
+  };
+
+  /**
+   * get all option elements (direct and nested children within `<bq-option-group>`)
+   */
+  private get optionItems(): HTMLBqOptionElement[] | [] {
+    if (!this.baseElem) return [];
+
+    const bqOptionItems: HTMLBqOptionElement[] = [];
+
+    this.slot.assignedElements({ flatten: true }).forEach((elem: HTMLBqOptionElement | HTMLBqOptionGroupElement) => {
+      if (isHTMLElement(elem, 'bq-option')) bqOptionItems.push(elem);
+      if (isHTMLElement(elem, 'bq-option-group'))
+        Array.from(elem.querySelectorAll('bq-option')).forEach((bqOption) => bqOptionItems.push(bqOption));
+    });
+
+    return bqOptionItems;
+  }
+
+  // render() function
+  // Always the last one in the class.
+  // ===================================
+
+  render() {
+    return (
+      <div ref={(element) => (this.baseElem = element)} part="base" role="listbox" aria-label="Options">
+        <slot onSlotchange={this.setSlotElem} />
+      </div>
```

Comment:
According to Figma, there's a gap space of 8px (spacing-xs2) between options. I think you should do the following here:

```tsx
      <div
        class="bq-option--list flex flex-col gap-y-xs"
        ref={(element) => (this.listElem = element)}
        part="base"
        role="listbox"
        aria-label={this.ariaLabel}
      >
        <slot />
      </div>
```

❗ The `aria-label` should not be hardcoded, please.